examples/all-clusters-app/all-clusters-common/include/tcc-mode.h

```diff
@@ -0,0 +1,74 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <app/clusters/mode-base-server/mode-base-server.h>
+#include <app/util/af.h>
+#include <app/util/config.h>
+#include <cstring>
+#include <utility>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+
+namespace RefrigeratorAndTemperatureControlledCabinetMode {
+
+const uint8_t ModeNormal      = 0;
+const uint8_t ModeRapidCool   = 1;
+const uint8_t ModeRapidFreeze = 2;
+
+/// This is an application level delegate to handle RefrigeratorAndTemperatureControlledCabinet commands according to the specific
+/// business logic.
+class TccModeDelegate : public ModeBase::Delegate
+{
+private:
+    using ModeTagStructType                     = detail::Structs::ModeTagStruct::Type;
+    ModeTagStructType modeTagsTccNormal[1]      = { { .value = to_underlying(ModeBase::ModeTag::kAuto) } };
```

Comment:
nit, i don't think you need the `1`
```suggestion
    ModeTagStructType modeTagsTccNormal[]      = { { .value = to_underlying(ModeBase::ModeTag::kAuto) } };
```